core/trino-main/src/main/java/io/trino/metadata/Catalog.java

```diff
@@ -66,12 +84,26 @@ public String getConnectorName()
         return connectorName;
     }
 
+    public boolean isFailed()
+    {
+        return catalogConnector == null;
+    }
+
+    public void verify()
+    {
+        if (catalogConnector == null) {
+            throw new TrinoException(CATALOG_UNAVAILABLE, format("Catalog '%s' failed to initialize and is disable", catalogName));
+        }
+    }
+
     public CatalogMetadata beginTransaction(
             TransactionId transactionId,
             IsolationLevel isolationLevel,
             boolean readOnly,
             boolean autoCommitContext)
     {
+        verify();
```

Comment:
If we don't plan to use this method elsewhere, inline it since it's only used here